From 876366b9e848b0b87c16fd22b7b4306a9879bb41 Mon Sep 17 00:00:00 2001
From: ran shalit <rshalit@habana.ai>
Date: Mon, 18 Oct 2021 22:40:01 +0300
Subject: [PATCH] [SW-56241] boot_info: add sha256 enable config

Add sha256 enable config, as preparation for enabling
sha256 only for specific devices.

Change-Id: Ibfadc53dd781f2fc52f77a6c4c3cc43a96f60b35
Signed-off-by: ran shalit <rshalit@habana.ai>
---

diff --git a/arch/gaudi/arm/common.h b/arch/gaudi/arm/common.h
index 69c7fc6..6964e28 100644
--- a/arch/gaudi/arm/common.h
+++ b/arch/gaudi/arm/common.h
@@ -43,6 +43,7 @@
 #define BTL_SYNC_TIMEOUT_US		(USEC_PER_SEC / 2)
 #define PCIE_EXT_LD_TIMEOUT_US		576
 
+#define SHA256_ENABLE
 
 #ifndef __ASSEMBLY__
 #include <spi.h>
diff --git a/arch/gaudi2/arm/common.h b/arch/gaudi2/arm/common.h
index 2653b83..7fcc703 100644
--- a/arch/gaudi2/arm/common.h
+++ b/arch/gaudi2/arm/common.h
@@ -120,5 +120,6 @@
 #define ARC_HALT_ACK_TIMEOUT_US				100
 #define MEM_CPY_TIMEOUT_US				(5 * USEC_PER_SEC)
 #define SP_SRAM_INIT_TIMEOUT_US				(5 * USEC_PER_SEC)
+
 #endif /* COMMON_GAUDI2_ARM_H_ */
 
diff --git a/arch/goya/arm/common.h b/arch/goya/arm/common.h
index e94fd08..18ca675 100644
--- a/arch/goya/arm/common.h
+++ b/arch/goya/arm/common.h
@@ -41,4 +41,6 @@
 #define PCIE_AUX_FLR_CTRL_INT_MASK_MASK		0x0
 #define mmPCIE_AUX_FLR_CTRL			0x0
 
+#define SHA256_ENABLE
+
 #endif /* COMMON_GOYA_H_ */
diff --git a/arch/greco/arc/common.h b/arch/greco/arc/common.h
index 6694722..28ce5b1 100644
--- a/arch/greco/arc/common.h
+++ b/arch/greco/arc/common.h
@@ -55,4 +55,6 @@
 #define ARC_RST_VALUE			0x110
 #define ARC_RST_VEC_ADDR		0x0
 
+#define SHA256_ENABLE
+
 #endif /* COMMON_GRECO_ARC_GNU_H_ */
diff --git a/arch/greco/arm/common.h b/arch/greco/arm/common.h
index 7ad082f..5b41909 100644
--- a/arch/greco/arm/common.h
+++ b/arch/greco/arm/common.h
@@ -57,4 +57,7 @@
 #define SW_ALL_RST_H_MSK		0x7FFFFF
 
 #define PCIE_WAIT_TIMEOUT_US		(5 * USEC_PER_SEC)
+
+#define SHA256_ENABLE
+
 #endif /* COMMON_GRECO_H_ */
diff --git a/lib/common/sha256.c b/lib/common/sha256.c
index 3677936..22e1222 100644
--- a/lib/common/sha256.c
+++ b/lib/common/sha256.c
@@ -11,6 +11,7 @@
 
 #define USE_BSWAP32
 
+#ifdef SHA256_ENABLE
 /*----------------------------------------------------------------*/
 
 const u8 sha256_der_prefix[SHA256_DER_LEN] = {
@@ -256,3 +257,12 @@
 	sha256_update(&ctx, input, ilen);
 	sha256_finish(&ctx, output);
 }
+
+#else
+
+void sha256_starts(struct sha256_context *ctx) {}
+void sha256_update(struct sha256_context *ctx, const u8 *input, u32 length) {}
+void sha256_finish(struct sha256_context *ctx, u8 digest[32]) {}
+void sha256_csum(const u8 *input, u32 ilen, u8 *output) {}
+
+#endif
